taichi/program/async_engine.cpp

```diff
@@ -132,31 +144,33 @@ void ExecutionQueue::enqueue(const TaskLaunchRecord &ker) {
     auto cloned_stmt = ker.ir_handle.clone();
     stmt = cloned_stmt->as<OffloadedStmt>();
 
-    compilation_workers.enqueue([async_func, stmt, kernel, this]() {
-      {
-        // Final lowering
-        using namespace irpass;
-
-        auto config = kernel->program.config;
-        auto ir = stmt;
-        offload_to_executable(
-            ir, config, /*verbose=*/false,
-            /*lower_global_access=*/true,
-            /*make_thread_local=*/true,
-            /*make_block_local=*/
-            is_extension_supported(config.arch, Extension::bls) &&
-                config.make_block_local);
-      }
-      auto func = this->compile_to_backend_(*kernel, stmt);
-      async_func->set(func);
-    });
+    compilation_workers.enqueue(
+        [kernel_name, async_func, stmt, kernel, this]() {
+          TI_TIMELINE(kernel_name);
```

Comment:
The only substantial change in the `compilation_workers.enqueue` call in this PR. The large diff here is due to braket removal.